src/main/java/internet/store/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,110 @@
+package internet.store.dao.jdbc;
+
+import internet.store.dao.ProductDao;
+import internet.store.exception.DataProcessingException;
+import internet.store.lib.Dao;
+import internet.store.model.Product;
+import internet.store.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+
+    @Override
+    public Product create(Product item) {
+        String query = "INSERT INTO products (product_name, price) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getProductName());
+            statement.setDouble(2, item.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+            return item;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Product "
+                    + item.toString() + " was not created", ex);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long itemId) {
+        String query = "SELECT * FROM products WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, itemId);
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            if (resultSet.getLong("deleted") == 0) {
```

Comment:
you can do it in the query string, like SELECT * FROM products WHERE id = ? AND delete = 0;